src/com/trollworks/gcs/character/TextTemplate.java

```diff
@@ -742,13 +858,43 @@ private void processHitLocationLoop(BufferedWriter out, String contents) throws
         }
     }
 
+    /* A kludgy method to relate a hitlocation to the armor that is providing the DR for that hit location. */
+    private String hitLocationEquipment(HitLocationTableEntry entry) {
+        StringBuilder sb    = new StringBuilder();
+        boolean       first = true;
+        for (Equipment equipment : mSheet.getCharacter().getEquipmentIterator()) {
+            if (equipment.isEquipped()) {
+                for (Feature feature : equipment.getFeatures()) {
+                    if (feature instanceof DRBonus) {
+                        String locationKey = entry.getLocation().getKey();
+                        if (locationKey.equals(HitLocation.VITALS.getKey())) {
+                            // Assume Vitals uses the same equipment as Torso
+                            locationKey = HitLocation.TORSO.getKey();
+                        }
+                        if (locationKey.endsWith(((DRBonus) feature).getLocation().name())) {
```

Comment:
You can and should be doing direct `==` comparison with the HitLocation -- they are all static final values.

Reply:
I cannot.   The HitLocation from the equipment is com.trollworks.gcs.character.HitLocation and the HitLocation from the DRBonus is com.trollworks.gcs.feature.HitLocation.   They are not the same object.    

The key for com.trollworks.gcs.character.HitLocation is created from strings from com.trollworks.gcs.character.Armor, which were created by prepending DR_PREFIX ("gcs.dr.") to the  com.trollworks.gcs.feature.HitLocation enum.    Which is why I had to go through such lengths to get them to match.   However, since the character.HitLocaiton name is built from the feature.HitLocation class, the string check will always be correct.

Reply:
Ah. I had forgotten about that. OK, ignore me! :-)